--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,8 @@ set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
 # Prefer building bundled dependencies as static instead of shared
 set(BUILD_SHARED_LIBS OFF)
 
+set(CMAKE_SYSTEM_NAME Linux)
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/externals/cmake-modules")
 include(DownloadExternals)
@@ -117,7 +119,7 @@ foreach(_opt IN LISTS _LIBRETRO_INCOMPATIBLE_OPTIONS)
 endforeach()
 
 option(ENABLE_SDL2 "Enable using SDL2" ON)
-option(USE_SYSTEM_SDL2 "Use the system SDL2 lib (instead of the bundled one)" OFF)
+option(USE_SYSTEM_SDL2 "Use the system SDL2 lib (instead of the bundled one)" ON)
 
 # Set bundled qt as dependent options.
 option(ENABLE_QT "Enable the Qt frontend" ON)
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -428,6 +428,7 @@ if (ENABLE_VULKAN)
     # spirv-headers
     if(USE_SYSTEM_SPIRV_HEADERS)
         find_package(SPIRV-Headers REQUIRED)
+        find_package(SPIRV-Tools REQUIRED)
         if(TARGET SPIRV-Headers::SPIRV-Headers)
             message(STATUS "Found SPIRV headers")
             get_target_property(SPIRV-Headers_SOURCE_DIR SPIRV-Headers::SPIRV-Headers INTERFACE_INCLUDE_DIRECTORIES)
@@ -438,10 +439,6 @@ if (ENABLE_VULKAN)
         add_subdirectory(spirv-headers EXCLUDE_FROM_ALL)
     endif()
 
-    # spirv-tools
-    # TODO: Implement USE_SYSTEM_SPIRV_TOOLS -OS
-    set(SPIRV_SKIP_EXECUTABLES ON)
-    add_subdirectory(spirv-tools EXCLUDE_FROM_ALL)
 
     # glslang
     if(USE_SYSTEM_GLSLANG)
--- a/externals/cmake-modules/CitraHandleSystemLibs.cmake
+++ b/externals/cmake-modules/CitraHandleSystemLibs.cmake
@@ -3,31 +3,31 @@ option(USE_SYSTEM_LIBS "Use system libraries over bundled ones" OFF)
 # System library options
 CMAKE_DEPENDENT_OPTION(USE_SYSTEM_QT "Use the system Qt lib (instead of the bundled one)" OFF "ENABLE_QT;MSVC OR APPLE" ON)
 CMAKE_DEPENDENT_OPTION(USE_SYSTEM_MOLTENVK "Use the system MoltenVK lib (instead of the bundled one)" OFF "APPLE" OFF)
-option(USE_SYSTEM_SDL2 "Use the system SDL2 lib (instead of the bundled one)" OFF)
-option(USE_SYSTEM_BOOST "Use the system Boost libs (instead of the bundled ones)" OFF)
-option(USE_SYSTEM_OPENSSL "Use the system OpenSSL libs (instead of the bundled LibreSSL)" OFF)
-option(USE_SYSTEM_LIBUSB "Use the system libusb (instead of the bundled libusb)" OFF)
+option(USE_SYSTEM_SDL2 "Use the system SDL2 lib (instead of the bundled one)" ON)
+option(USE_SYSTEM_BOOST "Use the system Boost libs (instead of the bundled ones)" ON)
+option(USE_SYSTEM_OPENSSL "Use the system OpenSSL libs (instead of the bundled LibreSSL)" ON)
+option(USE_SYSTEM_LIBUSB "Use the system libusb (instead of the bundled libusb)" ON)
 option(USE_SYSTEM_CPP_JWT "Use the system cpp-jwt (instead of the bundled one)" OFF)
 option(USE_SYSTEM_SOUNDTOUCH "Use the system SoundTouch (instead of the bundled one)" OFF)
 option(USE_SYSTEM_CPP_HTTPLIB "Use the system cpp-httplib (instead of the bundled one)" OFF)
-option(USE_SYSTEM_JSON "Use the system JSON (nlohmann-json3) package (instead of the bundled one)" OFF)
+option(USE_SYSTEM_JSON "Use the system JSON (nlohmann-json3) package (instead of the bundled one)" ON)
 option(USE_SYSTEM_DYNARMIC "Use the system dynarmic (instead of the bundled one)" OFF)
-option(USE_SYSTEM_FMT "Use the system fmt (instead of the bundled one)" OFF)
+option(USE_SYSTEM_FMT "Use the system fmt (instead of the bundled one)" ON)
 option(USE_SYSTEM_XBYAK "Use the system xbyak (instead of the bundled one)" OFF)
 option(USE_SYSTEM_OAKNUT "Use the system oaknut (instead of the bundled one)" OFF)
-option(USE_SYSTEM_INIH "Use the system inih (instead of the bundled one)" OFF)
+option(USE_SYSTEM_INIH "Use the system inih (instead of the bundled one)" ON)
 option(USE_SYSTEM_FFMPEG_HEADERS "Use the system FFmpeg headers (instead of the bundled one)" OFF)
-option(USE_SYSTEM_GLSLANG "Use the system glslang and SPIR-V libraries (instead of the bundled ones)" OFF)
+option(USE_SYSTEM_GLSLANG "Use the system glslang and SPIR-V libraries (instead of the bundled ones)" ON)
 option(USE_SYSTEM_ZSTD "Use the system Zstandard library (instead of the bundled one)" OFF)
-option(USE_SYSTEM_ENET "Use the system libenet (instead of the bundled one)" OFF)
-option(USE_SYSTEM_CRYPTOPP "Use the system cryptopp (instead of the bundled one)" OFF)
+option(USE_SYSTEM_ENET "Use the system libenet (instead of the bundled one)" ON)
+option(USE_SYSTEM_CRYPTOPP "Use the system cryptopp (instead of the bundled one)" ON)
 option(USE_SYSTEM_CUBEB "Use the system cubeb (instead of the bundled one)" OFF)
 option(USE_SYSTEM_LODEPNG "Use the system lodepng (instead of the bundled one)" OFF)
-option(USE_SYSTEM_OPENAL "Use the system OpenAL (instead of the bundled one)" OFF)
+option(USE_SYSTEM_OPENAL "Use the system OpenAL (instead of the bundled one)" ON)
 option(USE_SYSTEM_VMA "Use the system VulkanMemoryAllocator (instead of the bundled one)" OFF)
-option(USE_SYSTEM_VULKAN_HEADERS "Use the system Vulkan headers (instead of the bundled ones)" OFF)
-option(USE_SYSTEM_SPIRV_HEADERS "Use the system SPIRV headers (instead of the bundled ones)" OFF)
-option(USE_SYSTEM_CATCH2 "Use the system Catch2 (instead of the bundled one)" OFF)
+option(USE_SYSTEM_VULKAN_HEADERS "Use the system Vulkan headers (instead of the bundled ones)" ON)
+option(USE_SYSTEM_SPIRV_HEADERS "Use the system SPIRV headers (instead of the bundled ones)" ON)
+option(USE_SYSTEM_CATCH2 "Use the system Catch2 (instead of the bundled one)" ON)
 
 # Qt and MoltenVK are handled separately
 CMAKE_DEPENDENT_OPTION(DISABLE_SYSTEM_SDL2 "Disable system SDL2" OFF "USE_SYSTEM_LIBS" OFF)
--- a/externals/cubeb/CMakeLists.txt
+++ b/externals/cubeb/CMakeLists.txt
@@ -172,7 +172,6 @@ if(LAZY_LOAD_LIBS)
   check_include_files(alsa/asoundlib.h   USE_ALSA)
   check_include_files(jack/jack.h        USE_JACK)
   check_include_files(sndio.h            USE_SNDIO)
-  check_include_files(aaudio/AAudio.h    USE_AAUDIO)
 
   if(USE_PULSE OR USE_ALSA OR USE_JACK OR USE_SNDIO OR USE_AAUDIO)
     target_link_libraries(cubeb PRIVATE ${CMAKE_DL_LIBS})
@@ -213,7 +212,6 @@ else()
     target_link_libraries(cubeb PRIVATE sndio)
   endif()
 
-  check_include_files(aaudio/AAudio.h USE_AAUDIO)
   if(USE_AAUDIO)
     target_compile_definitions(cubeb PRIVATE DISABLE_LIBAAUDIO_DLOPEN)
     target_link_libraries(cubeb PRIVATE aaudio)
--- a/src/common/error.cpp
+++ b/src/common/error.cpp
@@ -6,6 +6,9 @@
 // Licensed under GPLv2 or any later version
 // Refer to the license.txt file included.
 
+#ifdef __TERMUX__
+#define _GNU_SOURCE 1
+#endif
 #include <cstddef>
 #ifdef _WIN32
 #include <windows.h>
@@ -35,7 +38,7 @@ std::string NativeErrorToString(int e) {
 #else
     char err_str[255];
 #if defined(__GLIBC__) && (_GNU_SOURCE || (_POSIX_C_SOURCE < 200112L && _XOPEN_SOURCE < 600)) ||   \
-    (defined(ANDROID) && !defined(HAVE_LIBRETRO))
+    (defined(ANDROID) && !defined(HAVE_LIBRETRO)) || defined(__TERMUX__)
     // Thread safe (GNU-specific)
     const char* str = strerror_r(e, err_str, sizeof(err_str));
     return std::string(str);
--- a/src/core/hle/service/soc/soc_u.cpp
+++ b/src/core/hle/service/soc/soc_u.cpp
@@ -1060,7 +1060,7 @@ void SOC_U::SockAtMark(Kernel::HLERequestContext& ctx) {
     func_res = ::ioctlsocket(holder.socket_fd, SIOCATMARK, &atMark);
     is_at_mark = atMark != 0;
 #else
-#ifdef ANDROID
+#if defined(ANDROID) || defined(__TERMUX__)
     func_res = 0;
     LOG_WARNING(Service_SOC, "(STUBBED) called");
 #else
